runner/app/live/pipelines/streamdiffusion.py

```diff
@@ -284,12 +279,38 @@ def _prepare_controlnet_configs(params: StreamDiffusionParams) -> Optional[List[
 
     return controlnet_configs
 
+def _prepare_ipadapter_configs(params: StreamDiffusionParams) -> Optional[Dict[str, Any]]:
+    """Prepare IPAdapter configurations for wrapper"""
+    if not params.ip_adapter:
+        return None
 
-def load_streamdiffusion_sync(params: StreamDiffusionParams, min_batch_size = 1, max_batch_size = 4, engine_dir = "engines", build_engines_if_missing = False):
-    # Prepare ControlNet configuration
-    controlnet_config = _prepare_controlnet_configs(params)
-    ipadapter_config = params.ip_adapter.model_dump() if params.ip_adapter else None
+    ip_cfg = params.ip_adapter.model_copy()
+    if ip_cfg.ipadapter_model_path:
+        logging.warning(f"[IPAdapter] ipadapter_model_path is deprecated and will be ignored. Use type instead.")
+    if ip_cfg.image_encoder_path:
+        logging.warning(f"[IPAdapter] image_encoder_path is deprecated and will be ignored. Use type instead.")
+
+    is_sdxl = 'sdxl' in params.model_id
+    arch = 'sdxl' if is_sdxl else 'sd15'
+    dir = 'sdxl_models' if is_sdxl else 'models'
+
+    if not ip_cfg.ipadapter_model_path:
+        match ip_cfg.type:
+            case 'regular':
+                ip_cfg.ipadapter_model_path = f"h94/IP-Adapter/{dir}/ip-adapter_{arch}.bin" # type: ignore
+            case 'faceid':
+                ip_cfg.ipadapter_model_path = f"h94/IP-Adapter-FaceID/ip-adapter-faceid_{arch}.bin" # type: ignore
+    if not ip_cfg.image_encoder_path:
+        ip_cfg.image_encoder_path = f"h94/IP-Adapter/{dir}/image_encoder" # type: ignore
```

Comment:
FTR had this whole logic implemented for custom ipadapter models for SDXL. Turns out SDXL+IPAdapt is not working on the base lib for now, but I'll keep the logic here as we should be fixing it soon.